src/render/painter.js

```diff
@@ -391,7 +391,7 @@ class Painter {
                 renderStencil(tileID);
             }
         } else {
-            if (this.nextStencilID + tileIDs.length > 256) {
+            if (Object.keys(this._tileClippingMaskIDs).length === 0 || this.nextStencilID + tileIDs.length > 256) {
```

Comment:
Nit: we could switch `_tileClippingMaskIDs` to `Map` too so that we can check if it's empty in constant time conveniently, instead of allocating a list of stringified tile keys first.

Reply:
Thanks for the suggestion, updated! 👌